plenum/test/view_change/test_view_change.py

```diff
@@ -72,3 +73,58 @@ def testViewChangeCase1(nodeSet, looper, up, wallet1, client1, viewNo):
     ensureElectionsDone(looper=looper, nodes=nodeSet)
     new_m_primary_node = get_master_primary_node(list(nodeSet.nodes.values()))
     assert m_primary_node.name != new_m_primary_node.name
+
+
+def test_view_change_timeout(nodeSet, looper, up, wallet1, client1, viewNo):
+    """
+    Check view change restarted if it is not completed in time
+    """
+
+    m_primary_node = get_master_primary_node(list(nodeSet.nodes.values()))
+
+    # Setting view change timeout to low value to make test pass quicker
+    for node in nodeSet:
+        node._view_change_timeout = 5
+
+    # Delaying view change messages to make first view change fail
+    # due to timeout
+    for node in nodeSet:
+        node.nodeIbStasher.delay(icDelay(delay=5))
+
+    # Delaying preprepae messages from nodes and
+    # sending request to force view change
+    for i in range(3):
+        delayNonPrimaries(nodeSet, instId=i, delay=10)
+    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, 4)
+
+    # First view change should fail, because of delayed
+    # instance change messages.
+    # This then leads to new view change that we need.
+    try:
```

Comment:
Correct, using try catch like this is wrong, you need to check that the exception is raised, here is no exception is raised, the test will work which is not what you want